### PR TITLE
move attrtibution-enabled SharedMap to experimental scope

### PR DESCRIPTION
**Pull request by clarenceli-msft**
Add the original `SharedMap` to `@fluid-experimental` package and remove the contents related to `SharedDirectory`, will add the attribution-related logic in next step.

### CODE REVIEW

File: experimental/dds/attributable-map/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# @fluid-experimental/attributable-map
+
```
Comment (Abe27342) [inline experimental/dds/attributable-map/README.md:2]: i'd add a note to the top of the readme that this is a fork of the "map" DDS which implements attribution, and longer-term plan is to remove it and integrate those APIs into map.

File: experimental/dds/attributable-map/src/test/types/validateMapPrevious.generated.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
```
Comment (Abe27342) [inline experimental/dds/attributable-map/src/test/types/validateMapPrevious.generated.ts:2]: let's just turn type validation off--delete this file and remove all of the scripts in package.json which relate to type validation. that might be why PR build is failing as well.

**Comment by tylerbutler** [comment]
I suggest rewording the PR title. It confused me a little before I looked at it closer. Maybe something like, "move SharedMap attribution to experimental scope"?

**Review by Abe27342 — APPROVED** [review]
